WebKit/WebKit
# [WebRTC] Add targets for vp8 and vp9 replay fuzzers

#### 6da809348ebef718d9d6bd9ae835afc5a482be2a
<pre>
[WebRTC] Add targets for vp8 and vp9 replay fuzzers
<a href="https://bugs.webkit.org/show_bug.cgi?id=263912">https://bugs.webkit.org/show_bug.cgi?id=263912</a>
&lt;<a href="https://rdar.apple.com/116139782">rdar://116139782</a>&gt;

Reviewed by Youenn Fablet.

* Source/ThirdParty/libwebrtc/Configurations/vp8_dec_fuzzer.xcconfig:
* Source/ThirdParty/libwebrtc/Configurations/vp9_dec_fuzzer.xcconfig:
- Drive-by fix to add license headers.
* Source/ThirdParty/libwebrtc/Configurations/vp8_replay_fuzzer.xcconfig: Add.
* Source/ThirdParty/libwebrtc/Configurations/vp9_replay_fuzzer.xcconfig: Add.
- Add xcconfig files for each fuzzer.

* Source/ThirdParty/libwebrtc/Source/third_party/json/include/json/json.h: Add.
(Json::CharReader::parse):
(Json::CharReaderBuilder::newCharReader):
- Temporary workaround for jsoncpp project.
* Source/ThirdParty/libwebrtc/Source/webrtc/test/call_config_utils.cc:
- Temporary workaround for jsoncpp project.

* Source/ThirdParty/libwebrtc/Source/webrtc/test/mac/video_renderer_mac.mm:
(-[CocoaWindow dealloc]): Add.
(-[CocoaWindow createWindow:]):
(-[CocoaWindow makeCurrentContext]):
(webrtc::test::MacRenderer::~MacRenderer):
(webrtc::test::MacRenderer::Init):
(webrtc::test::MacRenderer::OnFrame):
- Fix leaks caused by autoreleased objects getting added to an
  autoreleasePool of last resort and by failing to tear down objects in
  the -[CocoaWindow dealloc] method.

* Source/ThirdParty/libwebrtc/WebKit/0001-WebRTC-Add-targets-for-vp8-and-vp9-replay-fuzzers.patch: Add.

* Source/ThirdParty/libwebrtc/libwebrtc.xcodeproj/project.pbxproj:
- Add missing sources to the project for the vp{8,9}_replay_fuzzer
  targets.
- Add vp{8,9}_replay_fuzzer targets.

Canonical link: <a href="https://commits.webkit.org/270011@main">https://commits.webkit.org/270011@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/7e597e5a4d13e5824776593706f5e46e30619e9e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/24299 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/2409 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/14/builds/25386 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/26433 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/22368 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/4048 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/16/builds/24765 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/22813 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24543 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/1928 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/14/builds/25386 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/27023 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/1677 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/14/builds/25386 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/28136 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/22162 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/14/builds/25386 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/25929 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/1608 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/16/builds/24765 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/2885 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/14/builds/25386 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/2009 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3103 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/1976 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | |
